### PR TITLE
GH-32739: [CI][Docs] Document Docs PR Preview

### LINKED ISSUE #32739: [CI][Docs] Document Docs PR Preview
Document the changes from [ARROW-12958] here: https://arrow.apache.org/docs/developers/documentation.html

**Reporter**: [Jacob Wujciak](https://issues.apache.org/jira/browse/ARROW-17477) / @assignUser
**Watchers**: [Rok Mihevc](https://issues.apache.org/jira/browse/ARROW-17477) / @rok

<sub>**Note**: *This issue was originally created as [ARROW-17477](https://issues.apache.org/jira/browse/ARROW-17477). Please see the [migration documentation](https://github.com/apache/arrow/issues/14542) for further details.*</sub>

### PR DESCRIPTION
**Pull request by AlenkaF**
Add Pull Request Docs Preview section to developers/documentation.rst.
* Closes: #32739

### CODE REVIEW

**Review by raulcd — APPROVED** [review]
LGTM

File: docs/source/developers/documentation.rst [hunk 1]
```diff
@@ -102,6 +102,32 @@ The final output is located under the ``${PWD}/docs`` directory.
 
    :ref:`docker-builds`.
 
+Pull Request Docs Preview
+-------------------------
```
Comment (pitrou) [inline docs/source/developers/documentation.rst:106]:
```suggestion
Building a docs preview in a Pull Request
-----------------------------------------
```

File: docs/source/developers/documentation.rst [hunk 2]
```diff
@@ -102,6 +102,32 @@ The final output is located under the ``${PWD}/docs`` directory.
...
+
+You can use github-actions to build and host the documentation with crossbow within
+a pull request you are working on.
```
Comment (pitrou) [inline docs/source/developers/documentation.rst:109]:
```suggestion
You can build and preview the documentation within a GitHub pull request you are working on.
```

File: docs/source/developers/documentation.rst [hunk 3]
```diff
@@ -102,6 +102,32 @@ The final output is located under the ``${PWD}/docs`` directory.
...
+
+To use github-actions add a comment ``@github-actions crossbow submit preview-docs``
+to the pull request. The rendered documentation will then be available within the
+github-actions response, where you need to click on the crossbow build badge:
```
Comment (pitrou) [inline docs/source/developers/documentation.rst:113]:
```suggestion
To do so, post the comment ``@github-actions crossbow submit preview-docs``
to the pull request. The rendered documentation will then be available within the
GitHub Actions response, where you need to click on the Crossbow build badge:
```

**Review by pitrou — COMMENTED** [review]
Some minor suggestions. Also cc @jorisvandenbossche

**Review by jorisvandenbossche — APPROVED** [review]
Thanks, looks good!

**Pull request merged**